packages/react-dom/src/server/ReactPartialRenderer.js

```diff
@@ -689,14 +689,21 @@ class ReactDOMServerRenderer {
     this.providerStack[this.providerIndex] = null;
     this.providerIndex -= 1;
     const context: ReactContext<any> = provider.type._context;
-    if (this.providerIndex < 0) {
-      context._currentValue = context._defaultValue;
-    } else {
-      // We assume this type is correct because of the index check above.
-      const previousProvider: ReactProvider<any> = (this.providerStack[
-        this.providerIndex
-      ]: any);
+    // find the correct previous provider based on type
+    let previousProvider;
+    if (this.providerIndex > -1) {
+      for (let i = 0; i <= this.providerIndex; i += 1) {
+        if (this.providerStack[i] &&
```

Comment:
Please be explicit in type checks (e.g. `!== null` if that's what you mean). Don't rely on falsiness.

Reply:
OK. I wasn't sure if there was a more Flow appropriate way to check. Flow wanted explicit checks for both `null` and `undefined`.